src/jit/treelifeupdater.cpp

```diff
@@ -0,0 +1,366 @@
+#include "jitpch.h"
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+#include "treelifeupdater.h"
+
+template <bool ForCodeGen>
+TreeLifeUpdater<ForCodeGen>::TreeLifeUpdater(Compiler* compiler)
+    : compiler(compiler)
+    , newLife(VarSetOps::MakeEmpty(compiler))
+    , stackVarDeltaSet(VarSetOps::MakeEmpty(compiler))
+    , varDeltaSet(VarSetOps::MakeEmpty(compiler))
+    , gcTrkStkDeltaSet(VarSetOps::MakeEmpty(compiler))
+#ifdef DEBUG
+    , gcVarPtrSetNew(VarSetOps::MakeEmpty(compiler))
+    , epoch(compiler->GetCurLVEpoch())
+#endif // DEBUG
+{
+}
+
+//------------------------------------------------------------------------
+// UpdateLife: Determine does the tree affects liveness and processed accordingly.
```

Comment:
"Determine whether the tree affects liveness, and update liveness sets accordingly.

Reply:
Fixed, thanks.